pep-0570.rst

```diff
@@ -385,6 +385,202 @@ as a rule is not LL(1). This is the reason the rule needs to be included in
+
+
+There are more scenarios when positional-only parameters are beneficial. One of such
+scenarios is where a function can take some parameter and *any* keyword. For instance,
+let's imagine that we need a function that is able to format some template object. You
+can start to write ::
+
+    def format_template(template, **kwargs):
+        if isinstance(template, FastTemplate):
+            return template.format_fast(**kwargs)
+        elif isinstance(template, SlowTemplate):
+            return template.format_slow(**kwargs)
+        else:
+            return template.generic_format(**kwargs)
+
+Now you can call you function as ::
+
+    >>> format_template(slow_template, name="Hilbert", manifold="hausdorff")
+
+The problem with this approach is that you will never be able to pass the keyword
+"template" as the function will bind that value to the first argument ::
+
+    format_template(slow_template, name="Hilbert", template="fractal")
+    Traceback (most recent call last):
+      File "<stdin>", line 1, in <module>
+      TypeError: format_template() got multiple values for argument 'template'
+
+If you declare your function using positional-only parameters, this problem will go
+away as the first argument (template) is just a mere placeholder for the "template"
+object ::
+
+    def format_template(template, /, **kwargs):
+        if isinstance(template, FastTemplate):
+            return template.format_fast(**kwargs)
+        elif isinstance(template, SlowTemplate):
+            return template.format_slow(**kwargs)
+        else:
+            return template.generic_format(**kwargs)
+
+Now users can call your function with *all* possible keyword parameters ::
+
+    >>> format_template(slow_template, name="Hilbert", template="fractal")
+    "After some slow computations, *Hilbert* has finished the *fractal* template"
+
+Some of the properties of positional-only parameters can be emulated using
+``*args`` and some manual processing. This has the obvious downside that you
+need to do the processing on each function and that the signature of the
+functions that you create will not reflect how it must be called. For
+instance, for the last example, the signature will be ::
+
+    >>> help(format_template)
+    Help on function format_template in module best_functions:
+
+    format_template(*args, **kwargs)
+
+but with positional-only parameters will be ::
+
+    >>> help(format_template)
+    Help on function format_template in module best_functions:
+
+    format_template(template, /, **kwargs)
+
+The second version is more explicit and clear and can benefit from systems
+that can generate the documentation automatically by reading the function
+signatures. Additionally, if you plan to use type annotations for your
+functions, using a single parameter instead of ``*args`` can be easier to
+deal with.
+
+Some other scenarios where positional-only parameters can appear are when
```

Comment:
If the syntax is valid in pure python then I think this does not add much value.
The only benefit of explaining something to users here would be they “it is faster”.

Reply:
What I wanted is to remark that you need to use it if you are proving a fallback for a function that behaves like this and is implemented in C. Notice there are more ways of having this behaviour without the argument clinic.

Reply:
Again, I think this could be advanced use cases which could be split out as such.